mypy/checker.py

```diff
@@ -1643,14 +1643,15 @@ def erase_override(t: Type) -> Type:
                     if not is_subtype(original.arg_types[i],
                                       erase_override(override.arg_types[i])):
                         arg_type_in_super = original.arg_types[i]
+                        arg_node = override.definition.arguments[i+1]
```

Comment:
This doesn't seem quite right. It seems that you are passing this to forward the correct context, but 1) you are passing the next argument as context? 2) arguments aren't contexts. You probably want to make sure the context is set correctly to begin with.